addons/survey/views/survey_templates_statistics.xml

```diff
@@ -189,38 +191,44 @@
     </template>
 
     <template id="question_result_text" name="Question: text result (text_box, char_box)">
-        <table class="table table-hover table-sm" t-att-id="'survey_table_question_%d' % question.id">
-            <thead>
-                <tr>
-                    <th>#</th>
-                    <th>User Responses</th>
-                </tr>
-            </thead>
-            <tbody>
-                <t t-foreach="table_data" t-as="input_line">
-                    <tr t-if="not input_line.skipped" t-att-class="'d-none' if input_line_index >= page_record_limit else ''">
-                        <td>
-                            <t t-if="no_print_url"><t t-esc="input_line_index + 1"></t></t>
-                            <t t-else="">
-                                <a t-att-href="input_line.user_input_id.get_print_url()">
-                                    <t t-esc="input_line_index + 1"></t>
-                                </a>
-                            </t>
-                        </td>
-                        <td>
-                            <t t-if="question.question_type == 'text_box'">
-                                <span t-if="no_print_url" t-field="input_line.value_text_box" />
-                                <a t-else="" t-att-href="input_line.user_input_id.get_print_url()" t-field="input_line.value_text_box"></a>
-                            </t>
-                            <t t-if="question.question_type == 'char_box'">
-                                <span t-if="no_print_url" t-field="input_line.value_char_box" />
-                                <a t-else="" t-att-href="input_line.user_input_id.get_print_url()" t-field="input_line.value_char_box"></a>
-                            </t>
-                        </td>
+        <t t-set="non_skipped_records" t-value="list(filter(lambda record: not record.skipped, table_data))"/>
+        <t t-set="first_page_records_count" t-value="len(non_skipped_records) if len(non_skipped_records) &lt; page_record_limit else page_record_limit"/>
+        <t t-set="cell_height" t-value="default_line_height if question.question_type == 'char_box' else text_box_line_height"/>
+        <div t-attf-style="height: calc((#{cell_height}px * #{first_page_records_count}) + #{default_line_height}px + 16px);" class="overflow-auto">
+            <table class="table table-hover table-sm o_indexed_table" t-att-id="'survey_table_question_%d' % question.id">
+                <thead>
+                    <tr>
+                        <th>#</th>
+                        <th>User Responses</th>
                     </tr>
-                </t>
-            </tbody>
-        </table>
+                </thead>
+                <tbody>
+                    <t t-set="answered_input_index" t-value="[index for index, input_line in enumerate(table_data) if not input_line.skipped]"/>
+                    <t t-foreach="table_data" t-as="input_line">
```

Comment:
Would looping on our new "non_skipped_records" (see t-set above) achieve the same result without changing anything else?

We could even remove the `t-if="not input_line.skipped"` as we do not seem to care for skipped records here.